meteor/client/ui/Shelf/MiniRundownPanel.tsx

```diff
@@ -0,0 +1,241 @@
+import * as React from 'react'
+import * as _ from 'underscore'
+import ClassNames from 'classnames'
+import {
+	RundownLayoutBase,
+	DashboardLayoutMiniRundown,
+	RundownLayoutMiniRundown,
+	DashboardLayoutNextInfo,
+} from '../../../lib/collections/RundownLayouts'
+import { RundownLayoutsAPI } from '../../../lib/api/rundownLayouts'
+import { withTracker } from '../../lib/ReactMeteorData/ReactMeteorData'
+import { MeteorReactComponent } from '../../lib/MeteorReactComponent'
+import { RundownPlaylist } from '../../../lib/collections/RundownPlaylists'
+import { PartInstance, PartInstanceId, PartInstances } from '../../../lib/collections/PartInstances'
+import { Segment, Segments } from '../../../lib/collections/Segments'
+import { dashboardElementPosition } from './DashboardPanel'
+import { Part } from '../../../lib/collections/Parts'
+import { unprotectString } from '../../../lib/lib'
+interface IMiniRundownPanelProps {
+	key: string
+	visible?: boolean
+	layout: RundownLayoutBase
+	panel: RundownLayoutMiniRundown
+	playlist: RundownPlaylist
+}
+
+interface IMiniRundownPanelTrackedProps {
+	currentSegment?: Segment
+	currentPartInstance?: PartInstance
+	nextPartInstance?: PartInstance
+	nextSegment?: Segment
+}
+
+interface IState {}
+
+interface MiniRundownPart {
+	identifier: any
+	segmentName: string
+	partName: string
+}
+
+interface NextSegmentAndPart {
+	segment: Segment | undefined
+	part: Part | undefined
+}
+
+export class MiniRundownPanelInner extends MeteorReactComponent<
+	IMiniRundownPanelProps & IMiniRundownPanelTrackedProps,
+	IState
+> {
+	constructor(props) {
+		super(props)
+		this.state = {}
+	}
+
+	render() {
+		const isDashboardLayout = RundownLayoutsAPI.isDashboardLayout(this.props.layout)
+		const style = getElementStyle(this.props, isDashboardLayout)
+		const showAny = isShowAnyValues(this.props)
+
+		const current = getMiniRundownPart(
+			this.props,
+			showAny,
+			this.props.currentSegment,
+			this.props.currentPartInstance?.part
+		)
+		const next = getMiniRundownPart(this.props, showAny, this.props.nextSegment, this.props.nextPartInstance?.part)
+
+		const allParts: Part[] = this.props.playlist.getAllOrderedParts()
+		const allSegments: Segment[] = this.props.playlist.getSegments()
```

Comment:
I don't think these are reactive when placed in `render()`. They should be in tracker's computation, and they probably should be memoized.